compute/oslogin/service_account_ssh_test.py

```diff
@@ -37,66 +37,76 @@
-        'iam', 'v1', cache_discovery=False)
-    compute = googleapiclient.discovery.build(
-        'compute', 'v1', cache_discovery=False)
+    iam = googleapiclient.discovery.build("iam", "v1", cache_discovery=False)
+    compute = googleapiclient.discovery.build("compute", "v1", cache_discovery=False)
 
     # Create the necessary test resources and retrieve the service account
     # email and account key.
     try:
-        print('Creating test resources.')
+        print("Creating test resources.")
         service_account_key = setup_resources(
-            compute, iam, project, test_id, zone, image_family,
-            machine_type, account_email)
+            compute,
+            iam,
+            project,
+            test_id,
+            zone,
+            image_family,
+            machine_type,
+            account_email,
+        )
     except Exception:
-        print('Cleaning up partially created test resources.')
+        print("Cleaning up partially created test resources.")
         cleanup_resources(compute, iam, project, test_id, zone, account_email)
-        raise Exception('Could not set up the necessary test resources.')
+        raise Exception("Could not set up the necessary test resources.")
 
     # Get the target host name for the instance
-    hostname = compute.instances().get(
-        project=project,
-        zone=zone,
-        instance=test_id,
-        fields='networkInterfaces/accessConfigs/natIP'
-    ).execute()['networkInterfaces'][0]['accessConfigs'][0]['natIP']
+    hostname = (
+        compute.instances()
+        .get(
+            project=project,
+            zone=zone,
+            instance=test_id,
+            fields="networkInterfaces/accessConfigs/natIP",
+        )
+        .execute()["networkInterfaces"][0]["accessConfigs"][0]["natIP"]
+    )
 
     # Create a credentials object and use it to initialize the OS Login API.
     credentials = service_account.Credentials.from_service_account_info(
-        json.loads(base64.b64decode(
-            service_account_key['privateKeyData']).decode('utf-8')))
+        json.loads(
+            base64.b64decode(service_account_key["privateKeyData"]).decode("utf-8")
+        )
+    )
 
     oslogin = googleapiclient.discovery.build(
-        'oslogin', 'v1', cache_discovery=False, credentials=credentials)
-    account = 'users/' + account_email
+        "oslogin", "v1", cache_discovery=False, credentials=credentials
+    )
+    account = "users/" + account_email
 
     # More exceptions could be raised, keeping track of ones I could
     # find for now.
-    @backoff.on_exception(backoff.expo,
-                          (CalledProcessError,
-                           RefreshError),
-                          max_tries=5)
+    @backoff.on_exception(backoff.expo, (CalledProcessError, RefreshError), max_tries=5)
     def ssh_login():
         main(cmd, project, test_id, zone, oslogin, account, hostname)
         out, _ = capsys.readouterr()
```

Comment:
Instead of capturing output, have the function return a value/values, and test the returned value/values.